### ISSUE TITLE
`react/only-export-components` is reported against Storybook config files under `.storybook/`

### ISSUE THREAD

**Issue by wakamsha** [moneyforward/eslint-config-moneyforward#522]
## Summary

The `react` rule set configures [`react/only-export-components` as `warn`](https://github.com/moneyforward/frontend-tools/blob/main/packages/oxlint-config/src/rules/react.ts#L56). Storybook configuration files under `.storybook/` (`preview.tsx`, `preview.ts`, ...) legitimately export non-component values — `decorators`, `parameters`, `globalTypes` — next to JSX, so the rule reports files that are not React modules. This is a false positive: those files are read by Storybook, not by React Fast Refresh.

## Steps to reproduce

1. Set up a repository that has `.storybook/preview.tsx` exporting both a `decorators` array containing JSX and a default `Preview` object.
2. Configure oxlint with the rule sets a Storybook repository would use:

   ```ts
   import { defineConfig } from 'oxlint';
   import { essentials, react, storybook, typescript } from 'oxlint-config-moneyforward';

   export default defineConfig({
     extends: [essentials, typescript, react, storybook],
   });
   ```

3. Run `npx oxlint`.
4. Look at the diagnostics reported against `.storybook/preview.tsx`.

## Expected behavior

No diagnostic. The `**/.storybook/**/*.*` override in the `storybook` rule set should turn `react/only-export-components` off, the same way the `nextjs` rule set [already turns it off for route files](https://github.com/moneyforward/frontend-tools/blob/main/packages/oxlint-config/src/rules/nextjs.ts#L25).

## Proposed change

In [`packages/oxlint-config/src/rules/storybook.ts`](https://github.com/moneyforward/frontend-tools/blob/main/packages/oxlint-config/src/rules/storybook.ts), add the rule to the `**/.storybook/**/*.*` override:

```diff
   files: ['**/.storybook/**/*.*'],
   rules: {
     'import/no-default-export': ['off'],
+    'react/only-export-components': ['off'],
     'storybook/no-uninstalled-addons': ['error'],
   },
```

`packages/oxlint-config/src/configs/storybook/__snapshots__/snapshot.test.ts.snap` has to be regenerated along with it.

## Environment

- `oxlint-config-moneyforward`: 1.0.0
- `oxlint`: 1.78.0
- `oxlint-tsgolint`: 7.0.2001
- Storybook: any version that uses a `.storybook/preview.tsx` with decorators

## Others

- The maintainer already has this change locally; it will be included in the batch pull request for v1.1.0. Since `react/only-export-components` is a `warn` rule, this change is a **minor** bump under [the versioning policy](https://github.com/moneyforward/frontend-tools/tree/main/packages/oxlint-config#versioning).
- **Open question**: the sibling `**/*.{stories,story}.*` override does *not* disable this rule. Story files export a default `Meta` plus named `StoryObj` values, so they may need the same treatment — needs confirmation against a real repository before deciding.
